src/commands/extractApis.ts

```diff
@@ -0,0 +1,114 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as fse from 'fs-extra';
+import * as path from 'path';
+import { OpenDialogOptions, ProgressLocation, Uri, window, workspace } from "vscode";
+import { AzureTreeItem } from 'vscode-azureextensionui';
+import * as Constants from '../constants';
+import { ApiTreeItem } from '../explorer/ApiTreeItem';
+import { ext } from '../extensionVariables';
+import { LinkedApisJsonObject } from "../extractApi/LinkedApisJsonObject";
+import { SingleApiJsonObject } from "../extractApi/SingleApiJsonObject";
+import { localize } from '../localize';
+import { cpUtils } from '../utils/cpUtils';
+import { getDefaultWorkspacePath } from '../utils/fsUtil';
+
+export async function extractApis(node: AzureTreeItem | ApiTreeItem): Promise<void> {
+    const uris = await askFolder();
+    const templatesFolder = await createTemplatesFolder(uris);
+    let configFile: string = "";
+    let noticeContent: string = "";
+
+    if (node instanceof ApiTreeItem) {
+        const apiNode: ApiTreeItem = node;
+
+        const sourceApimName = apiNode.root.serviceName;
+        const resourceGroup = apiNode.root.resourceGroupName;
+        const apiName = apiNode.apiContract.name == null ? "" : apiNode.apiContract.name;
+
+        const singleApiJson = genSingleApiJsonObj(sourceApimName, resourceGroup, templatesFolder, apiName);
+
+        configFile = uris[0].fsPath.concat("/", apiName, ".json");
+        noticeContent = `Extract API '${apiName}' to '${templatesFolder}'`;
+        await fse.writeFile(configFile, singleApiJson);
+    } else if (node instanceof AzureTreeItem) {
+        const serviceNode: AzureTreeItem = node;
+
+        const fullId = serviceNode.id === undefined ? "" : serviceNode.id;
+        const apiParams = fullId.split("/");
+
+        //generate linked templates
+        const linkedApiJson = genLinkedApisJsonObj(apiParams[8], apiParams[4], templatesFolder);
+
+        configFile = uris[0].fsPath.concat(Constants.extractorMasterJsonFile);
```

Comment:
Use serviceName instead of constant so that user can run extract command on more than one service.